torch/distributed/distributed_c10d.py

```
@@ -4995,18 +4995,23 @@ def split_group(
 
     # if the parent backend does not support splitting, raise error
     # currently this API only support NCCL backend
-    if (
-        not parent_backend
-        or not parent_backend.supports_splitting
-        or not isinstance(parent_backend, ProcessGroupNCCL)
-    ):
+    if not parent_backend or not parent_backend.supports_splitting:
         raise RuntimeError(
             "No backend for the parent process group or its backend does not support splitting"
         )
 
+    missing_methods = [
+        method
+        for method in ["comm_split_count", "perform_nocolor_split"]
+        if not hasattr(parent_backend, method)
+    ]
+    assert not missing_methods, (
+        f"Backend is missing required methods for split operation: {missing_methods}"
+    )
+
```

nit: no need? we can let the program error out later.